src/main/java/ejercicio/banco/service/BankServiceImpl.java

```
@@ -14,8 +14,14 @@ public BankServiceImpl(BankRepository repository) {
     }
 
     public Bank processBank(String filename) throws IndexOutOfBoundsException{
+        if (filename == null) {
+            throw new IllegalArgumentException("Filename must not be null");
+        }
         Bank bank;
         List<Bank> banks = repository.findAll(filename);
+        if(banks.size() < 1){
+            return null;
```

If we couldn't find any Bank by the filename I would throw an exception here.